File: myheroesinkotlin/app/src/main/res/drawable/app/src/main/java/com/example/myheroes/veiwmodel/HeroViewModel.java
package com.example.myheroes.veiwmodel;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.example.myheroes.model.HeroResults;
import com.example.myheroes.network.RetrofitInstance;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class HeroViewModel extends AndroidViewModel {

    private RetrofitInstance retrofitInstance = new RetrofitInstance();

    private MutableLiveData<HeroResults> resultLiveData = new MutableLiveData<>();


    public HeroViewModel(@NonNull Application application) {

        super( application );

    }


    public void getHeroRepo() {

        retrofitInstance.getRepositories().enqueue( new Callback<HeroResults>() {

            @Override
            public void onResponse(Call<HeroResults> call, Response<HeroResults> response) {
                Log.d( "TAG", "onResponse: " );
                resultLiveData.setValue( response.body() );

            }

            @Override
            public void onFailure(Call<HeroResults> call, Throwable t) {

            }

        } );
    }

    public void getHeroRepo(String hero) {

        retrofitInstance.getRepositories( hero ).enqueue( new Callback<HeroResults>() {

            @Override
            public void onResponse(Call<HeroResults> call, Response<HeroResults> response) {

                resultLiveData.setValue( response.body() );

            }

            @Override
            public void onFailure(Call<HeroResults> call, Throwable t) {
            }

        } );
    }

    public MutableLiveData<HeroResults> getResultLiveData() {

        return resultLiveData;

    }
}

File: myheroesinkotlin/app/src/main/java/com/example/myheroesinkotlin/viewmodel/HeroViewModel.kt
package com.example.myheroesinkotlin.viewmodel

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.example.myheroesinkotlin.model.HeroResults
import com.example.myheroesinkotlin.network.RetrofitInstance
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class HeroViewModel (application: Application) : AndroidViewModel(application){
    private var retrofitInstance = RetrofitInstance()
    private var resultLiveData = MutableLiveData<HeroResults>()


    fun getHeroRepo(){
        retrofitInstance.getRepositories()?.enqueue(object : Callback<HeroResults?> {
            override fun onResponse(
                call: Call<HeroResults?>,
                response: Response<HeroResults?>
            ) {
                Log.d("TAG", "onResponse: ")
                resultLiveData.value = response.body()
            }

            override fun onFailure(
                call: Call<HeroResults?>,
                t: Throwable
            ) {
            }
        })

    }


    fun getHeroRepo(hero: String) {
        retrofitInstance.getRepositories(hero)?.enqueue(object : Callback<HeroResults?> {
            override fun onResponse(
                call: Call<HeroResults?>,
                response: Response<HeroResults?>
            ) {
                resultLiveData.value = response.body()
            }

            override fun onFailure(
                call: Call<HeroResults?>,
                t: Throwable
            ) {
            }
        })
    }

    fun getResultLiveData () : MutableLiveData<HeroResults>{
        return resultLiveData
    }

}